R/resource.R

```diff
@@ -280,21 +289,32 @@ hipercow_cluster_info <- function(driver = NULL, root = NULL) {
 ##'   error = identity)
 ##'
 ##' cleanup()
-hipercow_resources_validate <- function(resources,
-                                        driver = NULL,
-                                        root = NULL) {
-
+hipercow_resources_validate <- function(resources, driver = NULL, root = NULL) {
   root <- hipercow_root(root)
-  # A bit of a hack for now - handle NULL driver (which often happens
-  # in the tests) - and also ignore if more than one driver is
-  # configured, which is caught elsewhere; here we would return a
-  # "driver must be a scalar" error.
+  driver <- hipercow_driver_select(driver, FALSE, root, rlang::current_env())
+  resources_validate(resources, driver, root)
+}
 
-  if (is.null(driver) || length(driver) > 1) {
-    return(resources)
+
+resources_validate <- function(resources, driver, root) {
+  given_resources <- !is.null(resources)
+  if (given_resources) {
+    assert_is(resources, "hipercow_resource")
+    if (identical(attr(resources, "validated", exact = TRUE), driver)) {
```

Comment:
```suggestion
    if (!is.null(driver) && 
        (identical(attr(resources, "validated", exact = TRUE), driver))) {
```

Reply:
Otherwise, if attr(resources, "validated") is NULL (ie, it wasn't validated), then this can match when driver is NULL as well, and we return early, rather than erroring because of resources specified but no driver.